pkg/streamingpromql/operator/binary_operation.go

```diff
@@ -153,47 +153,44 @@ func (b *BinaryOperation) loadSeriesMetadata(ctx context.Context) (bool, error)
 // - a list indicating which series from the left side are needed to compute the output
 // - a list indicating which series from the right side are needed to compute the output
 func (b *BinaryOperation) computeOutputSeries() ([]SeriesMetadata, []*binaryOperationOutputSeries, []bool, []bool) {
-	// TODO: Prometheus' engine uses strings for the key here, which would avoid issues with hash collisions, but seems much slower.
-	// Either we should use strings, or we'll need to deal with hash collisions.
-	hashFunc := b.hashFunc()
-	outputSeriesMap := map[uint64]*binaryOperationOutputSeries{}
+	labelsFunc := b.labelsFunc()
+	outputSeriesMap := map[string]*binaryOperationOutputSeries{}
 
 	// TODO: is it better to use whichever side has fewer series for this first loop? Should result in a smaller map and therefore less work later on
 	// Would need to be careful about 'or' and 'unless' cases
 	for idx, s := range b.leftMetadata {
-		hash := hashFunc(s.Labels)
-		series, exists := outputSeriesMap[hash]
+		groupLabels := labelsFunc(s.Labels).String()
```

Comment:
is labelsFunc guaranteed to be sorted?

Reply:
Yes, `labels.Labels` guarantees this, at least for the methods we're using.